### ISSUE TITLE
Build fails with LLVM flang 19

### ISSUE THREAD

**Issue by awvwgk** [fortran-lang/fpm#1091]
Looks like flang cannot compile fpm 0.10.1 at the moment. The build of the bootstrapper fails with multiple errors.

See https://github.com/conda-forge/fpm-feedstock/pull/27

**Comment by awvwgk** [comment]
Error message from building the bootstrapper, unfortunately not mapped to the fpm source files:

````
(%BUILD_PREFIX%) %SRC_DIR%>flang-new -J build\bootstrap -o build\bootstrap\fpm.exe build\bootstrap\fpm.F90 
error: Semantic errors in build\bootstrap\fpm.F90
.\build\bootstrap\fpm.F90:27433:14: error: 'toml_error' from host is not accessible
          type(toml_error), allocatable :: toml_error
               ^^^^^^^^^^
.\build\bootstrap\fpm.F90:27433:42: because: 'toml_error' is hidden by this entity
          type(toml_error), allocatable :: toml_error
                                           ^^^^^^^^^^
.\build\bootstrap\fpm.F90:28256:22: error: Override of PRIVATE DEFERRED 'serializable_is_same' must appear in its module
          procedure :: serializable_is_same => git_is_same
                       ^^^^^^^^^^^^^^^^^^^^
.\build\bootstrap\fpm.F90:27293:24: Procedure 'serializable_is_same' of type 'serializable_t' is bound to 'is_equal'
        logical function is_equal(this,that)
                         ^^^^^^^^
.\build\bootstrap\fpm.F90:28257:22: error: Override of PRIVATE DEFERRED 'dump_to_toml' must appear in its module
          procedure :: dump_to_toml
                       ^^^^^^^^^^^^
.\build\bootstrap\fpm.F90:27261:18: Procedure 'dump_to_toml' of type 'serializable_t' is bound to 'to_toml'
        subroutine to_toml(self, table, error)
                   ^^^^^^^
.\build\bootstrap\fpm.F90:28258:22: error: Override of PRIVATE DEFERRED 'load_from_toml' must appear in its module
          procedure :: load_from_toml
                       ^^^^^^^^^^^^^^
.\build\bootstrap\fpm.F90:27277:18: Procedure 'load_from_toml' of type 'serializable_t' is bound to 'from_toml'
        subroutine from_toml(self, table, error)
                   ^^^^^^^^^
.\build\bootstrap\fpm.F90:28263:15: error: Generic 'OPERATOR(==)' may not have specific procedures 'serializable_t%serializable_is_same' and 'git_target_eq' as their interfaces are not distinguishable
      interface operator(==)
                ^^^^^^^^^^^^
````

**Comment by perazz** [comment]
flang compilation was fixed in #1011 #1013 #1024 #1028 #1029, trunk builds just fine so I will close this one. For the package managers, a new release may be needed.
